src/system/sys_profiler.cc

```diff
@@ -425,13 +582,26 @@ std::shared_ptr<std::ostream> Profiler::get_default_output_stream() {
 
 
 #ifdef FLOW123D_HAVE_MPI
+template <class T>
+void save_mpi_metric (property_tree::ptree &node, MPI_Comm comm, T * ptr, string name) {
+    node.put (name+"-min", MPI_Functions::min(ptr, comm));
+    node.put (name+"-max", MPI_Functions::max(ptr, comm));
+    node.put (name+"-sum", MPI_Functions::sum(ptr, comm));
+}
+
 void Profiler::output(MPI_Comm comm, ostream &os) {
     int ierr, mpi_rank, mpi_size;
     //wait until profiling on all processors is finished
     MPI_Barrier(comm);
     stop_timer(0);
+    propagate_timers();
+    
+    // stop monitoring memory
+    bool memory_monitoring_ = get_memory_monitoring();
```

Comment:
This is completely against our naming convention having the ending underscore for class members.